source/main.cpp

```diff
@@ -384,6 +384,15 @@ int main(int argc, char *argv[])
      * features.
      */
 #if !defined(DISABLE_MQTT)
+    /**
+     * init() is currently responsible for making sure only 1 instance of Device Client is running at a given time.
```

Comment:
why do we have to initialize twice?

Reply:
the first initialization is under the condition `if (config.config.fleetProvisioning.enabled &&
        !config.config.fleetProvisioningRuntimeConfig.completedFleetProvisioning)`